codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/UnionGenerator.java

```diff
@@ -109,7 +109,7 @@ public void generateUnionExamples(GoWriter writer) {
                             symbol.getNamespace()).build();
 
                     writer.openBlock("case *$T:", "", memberSymbol, () -> {
-                        writer.write("_ = v.Value // Value is $L", targetSymbol.getName());
+                        writer.write("_ = v.Value // Value is $T", targetSymbol);
```

Comment:
nit: Think `$L ` would have worked correctly if you used just `targetSymbol` instead of getName

Reply:
`$L` will write and the full import path + types, `github.com/aws/aws-sdk-go-v2/service/foo/types#SomeType`, which arguably is never the correct behavior, but its what $L does. Should look at updating $L to do "right" thing.